ViewModels/AnalysisViewModel.cs

```diff
@@ -0,0 +1,811 @@
+                _currentLibrary = "Wasatch";
+
+
+            settings.LibraryChanged += Settings_LibraryChanged;
+            Spectrometer.NewConnection += handleNewSpectrometer;
+            getInstance().PropertyChanged += AnalysisViewModel_PropertyChanged;
+            getInstance().SpectraChanged += AnalysisViewModel_SpectraChanged;
+        }
+
+        public AnalysisViewModel(bool isInstance = false)
+        {
+            spec = BluetoothSpectrometer.getInstance();
+            if (spec == null || !spec.paired)
+                spec = API6BLESpectrometer.getInstance();
+            if (spec == null || !spec.paired)
+                spec = USBSpectrometer.getInstance();
+
+            shareCmd = new Command(() => { _ = ShareSpectrum(); });
+            addCmd = new Command(() => { _ = doAdd(); });
+            correctionCmd = new Command(() => { _ = changeCorrection(); });
+            retryCmd = new Command(() => { _ = triggerReanalyze(); });
+            precisionCmd = new Command(() => { _ = triggerPrecision(); });
+
+            SetData(null, null);
+
+            Spectrometer.NewConnection += handleNewSpectrometer;
+            if (!isInstance)
+            {
+                getInstance().PropertyChanged += AnalysisViewModel_PropertyChanged;
+                getInstance().SpectraChanged += AnalysisViewModel_SpectraChanged;
+            }
+        }
+
+        public Command shareCmd { get; private set; }
+        public Command addCmd { get; private set; }
+        public Command saveCmd { get; private set; }
+        public Command correctionCmd { get; private set; }
+        public Command retryCmd { get; private set; }
+        public Command precisionCmd { get; private set; }
+
+        bool triggerReanalyze()
+        {
+            TriggerRetry?.Invoke(this, this);
+            return true;
+        }
+
+        bool triggerPrecision()
+        {
+            TriggerIncreasedPrecision?.Invoke(this, this);
+            return true;
+        }
+
+        private void AnalysisViewModel_SpectraChanged(object sender, AnalysisViewModel e)
+        {
+            updateFromInstance();
+        }
+
+        Dictionary<string, AutoRamanParameters> parameterSets = new Dictionary<string, AutoRamanParameters>()
+        {
+            {
+                "Default" ,
+                new AutoRamanParameters()
+                {
+                    maxCollectionTimeMS = 10000,
+                    startIntTimeMS = 100,
+                    startGainDb = 0,
+                    minIntTimeMS = 10,
+                    maxIntTimeMS = 2000,
+                    minGainDb = 0,
+                    maxGainDb = 30,
+                    targetCounts = 45000,
+                    minCounts = 40000,
+                    maxCounts = 50000,
+                    maxFactor = 5,
+                    dropFactor = 0.5f,
+                    saturationCounts = 65000,
+                    maxAverage = 100
+                }
+            },
```

Comment:
also defined in SettingsViewModel?

Reply:
this has ping ponged back and forth repeatedly as we finalized design. I'll try to unwrap in 1.0...I'll leave a comment for now

Reply:
one feature is tied into and another in to the other and one of those features was an 11th hour add, I don't feel like unwrapping it today...